exporter/chronicleforwarderexporter/exporter.go

```diff
@@ -0,0 +1,118 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chronicleforwarderexporter
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type chronicleForwarderExporter struct {
+	cfg       *Config
+	logger    *zap.Logger
+	writer    io.Writer
+	marshaler logMarshaler
+	endpoint  string
+}
+
+func newExporter(cfg *Config, params exporter.CreateSettings) (*chronicleForwarderExporter, error) {
+	exporter := &chronicleForwarderExporter{
+		cfg:       cfg,
+		logger:    params.Logger,
+		marshaler: newMarshaler(*cfg, params.TelemetrySettings),
+	}
+
+	switch cfg.ExportType {
+	case ExportTypeSyslog:
+		endpoint := buildEndpoint(cfg)
+
+		var conn net.Conn
+		var err error
+		if cfg.Syslog.TLSSetting != nil {
+			tlsConfig, err := cfg.Syslog.TLSSetting.LoadTLSConfig()
+			if err != nil {
+				return nil, fmt.Errorf("load TLS config: %w", err)
+			}
+			conn, err = tls.Dial(cfg.Syslog.Network, endpoint, tlsConfig)
+		} else {
+			conn, err = net.Dial(cfg.Syslog.Network, endpoint)
+		}
```

Comment:
This seems a little brittle to me, what happens if the connection is broken for a second, but could be re-established later? I think because we only dial once, any network failure would end up being permanent until the collector restarts.

Reply:
would it be better to dial on every log push?

Reply:
Yeah, it looks like that's what the upstream syslog exporter does
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c49dbc1e3f7189abb22e64fc26b72b0c896d1385/exporter/syslogexporter/exporter.go#L98

Reply:
reworked it to dial during logs push